lms/services/canvas_studio.py

```diff
@@ -130,9 +151,14 @@ def authorization_url(self, state: str) -> str:
 
         return auth_url
 
+    def _token_url(self) -> str:
+        """Return the URL of the Canvas Studio OAuth token endpoint."""
+        return self._api_url("oauth/token")
+
     def redirect_uri(self) -> str:
         """Return OAuth redirect URI for Canvas Studio."""
-        return self._request.route_url("canvas_studio_api.oauth.callback")
+        redirect_uri = self._request.route_url("canvas_studio_api.oauth.callback")
+        return replace_localhost_in_url(redirect_uri)
```

Comment:
In the context of OAuth token refresh, the `/api/canvas_studio/oauth/refresh` endpoint will be called with host URL `http://localhost:8001` in development, but we need to replace this when generating the request to Canvas Studio's OAuth token endpoint.